contracts/DutchExchange.sol

```diff
@@ -817,7 +748,7 @@ contract DutchExchange {
         // (Takes in a amount of user's TUL tokens as ration of all TUL tokens, outputs fee ratio)
         // We premultiply by amount to get fee:
         if (totalTUL > 0) {
-            feeRatio.num = totalTUL - 10 * balanceOfTUL;
+            feeRatio.num = Math.atleastZero(int(totalTUL - 10 * balanceOfTUL));
             feeRatio.den = 16000 * balanceOfTUL + 200 * totalTUL;
```

Comment:
`uint balanceOfTUL = TokenTUL(TUL).lockedTULBalances(user);` from a couple lines above can be moved inside `if (totalTUL > 0) {`

Would save a function call in case of `else`

Reply:
yyes